src/main/java/conan/commands/task/AsyncConanTask.java

```diff
@@ -47,17 +48,30 @@ public void run(@NotNull ProgressIndicator indicator) {
         try {
             String message = args.getCommandLineString();
             log(logger, message, "", NotificationType.INFORMATION);
-            ProcessHandler processHandler = new OSProcessHandler(args);
-            processHandler.startNotify();
-            if (processListener != null) {
-                processHandler.addProcessListener(processListener);
-            } else if (cmakeListener != null) {
-                cmakeListener.processStarted(processHandler);
-            } else if (getProject() != null){
-                ConanToolWindow conanToolWindow = ServiceManager.getService(getProject(), ConanToolWindow.class);
-                conanToolWindow.attachConsoleToProcess(processHandler, message, conanProfile);
+            BaseProcessHandler processHandler = new OSProcessHandler(args);
+
+            Runnable runnable = () -> OSProcessUtil.killProcessTree(processHandler.getProcess());
+            ShutDownTracker.getInstance().registerShutdownTask(runnable);
+
+            try {
+                if (processListener != null) {
+                    processHandler.addProcessListener(processListener);
+                } else if (cmakeListener != null) {
+                    cmakeListener.processStarted(processHandler);
+                } else if (getProject() != null) {
+                    ConanToolWindow conanToolWindow = ServiceManager.getService(getProject(), ConanToolWindow.class);
+                    conanToolWindow.attachConsoleToProcess(processHandler, "Running Conan...", conanProfile);
+                }
+                ProcessOutput processOutput = CidrToolsUtil.runWithProgress(processHandler, 0);
+                if (processOutput.isCancelled()) {
+                    throw new ProcessCanceledException();
+                }
             }
-            processHandler.waitFor();
+            finally {
+                ShutDownTracker.getInstance().unregisterShutdownTask(runnable);
+                runnable.run();
```

Comment:
I this line really necessary? The process should be terminated before the finally clause.

Reply:
If something fails inside the `try` block and the process is not stopped, this line will kill it. If the process finishes properly, then this line will do nothing (killing non-existing process). In fact, I just copied these from the disassembly of the CMakeRunner 😅